### PR TITLE
Update to ipywidgets 8 / Notebook 7 / JupyterLab 4

### PR DESCRIPTION
**Pull request by martinRenou**
Builds against https://github.com/twosigma/beakerx_base/pull/10 and https://github.com/twosigma/beakerx_kernel_groovy/pull/17

### CODE REVIEW

**Comment by martinRenou** [comment]
This does not work yet. There is some CodeMirror tweaking logic that does not work with JupyterLab 4 (CodeMirror 6).

This will require some days of work to get it right. Also we might want to consider dropping JupyterLab <=3 support, because supporting multiple CodeMirror versions may be tricky.

**Comment by martinRenou** [comment]
Screenshot from 2024-01-03 10-25-25

This is now working nicely.

We need to update the beakerx_test repository in order for the e2e tests to pass.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -47,11 +47,13 @@ jobs:
       uses: actions/checkout@v2
       with:
         repository: twosigma/beakerx_base
+        ref: master
         path: beakerx_base
     - name: Checkout beakerx_kernel_groovy repo
       uses: actions/checkout@v2
       with:
-        repository: twosigma/beakerx_kernel_groovy
+        repository: martinRenou/beakerx_kernel_groovy
```
Comment (TanayParikh) [inline .github/workflows/main.yml:55]: Please revert

**Comment by achhina** [comment]
@martinRenou would this warrant a bump in version number?

**Comment by martinRenou** [comment]
Compatibility should be kept for JupyterLab 3.

As long as you built the wheel in an environment where JupyterLab 4 lives, the resulting wheel can be installed in an environment where JupyterLab 3 is installed.

So I guess a 2.4.0 bump is enough.

**Comment by martinRenou** [comment]
@achhina I think this is ready for merging. We can bump the version number separately.

**Review by achhina — APPROVED** [review]

**Review by TanayParikh — APPROVED** [review]

**Pull request merged**